gobblin-temporal/src/main/java/org/apache/gobblin/temporal/workflows/metrics/TemporalEventTimer.java

```diff
@@ -70,7 +70,9 @@ private static Instant getCurrentTime() {
   }
 
   public static class Factory {
-    private static final ActivityOptions DEFAULT_OPTS = ActivityOptions.newBuilder().build();
+    private static final ActivityOptions DEFAULT_OPTS = ActivityOptions.newBuilder()
+        .setStartToCloseTimeout(Duration.ofHours(24))
```

Comment:
what are the ramifications of this?  a job could still run >24 hours right?  does this merely constrain the activity that attempts to send GTEs that it must finish its work of performing the send within 24 hours?

Reply:
Correct. the job itself can still run for > 24 hours. But if for some reason kafka is down, this operation can take an arbitrary amount of time.

There are 2 things to consider here:
1. GaaS won't be able to detect if a job has finished until this activity succeeds. If Kafka is down, we want to be able to retry this task until kafka is back.
2. If the job itself is completed but this timer is unable to send the message, I think 24 hours is a reasonable amount of time to retry for before giving up. We ideally don't want to have to retry the work just because the timer failed for a brief period of time (e.g. a few hours of downtime)

Reply:
sure that's fine.  for this, even 8 or 12 hours should be far sufficient.  please add clarifying comment that this is max time to attempt to send, potentially waiting out a kafka outage